From 9fccd03fb65e18b697835e2b83a33fee3bddf404 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 03 Aug 2021 15:56:41 +0300
Subject: [PATCH] [SW-0] gaudi: security: rework mmu_up and stlb

Rework the STLB and MMU_UP protection - switch to close all, keep open
only what is needed (used).
This is following the penetration test and discovery of registers that
may result in instability.

Change-Id: I7ad523f0479b2e33c2ee6e6e874295b31bd35614
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index dabaa91..a92de3a 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -873,16 +873,16 @@
 
 	/* blocking regs within offsets: 0 - 0x80 */
 	prot_base = base;
-	mask = (PROT_BIT(mmMMU_UP_FEATURE_ENABLE) &
-		PROT_BIT(mmMMU_UP_VA_ORDERING_MASK_31_7) &
-		PROT_BIT(mmMMU_UP_VA_ORDERING_MASK_49_32) &
-		PROT_BIT(mmMMU_UP_LOG2_DDR_SIZE) &
-		PROT_BIT(mmMMU_UP_SCRAMBLER) &
-		PROT_BIT(mmMMU_UP_MEM_INIT_BUSY) &
-		PROT_BIT(mmMMU_UP_DBG_MEM_WRAP_RM) &
-		PROT_BIT(mmMMU_UP_PIPE_CREDIT) &
-		PROT_BIT(mmMMU_UP_SLICE_CREDIT) &
-		PROT_BIT(mmMMU_UP_MMU_BYPASS));
+	mask = ~(PROT_BIT(mmMMU_UP_MMU_ENABLE) &
+		 PROT_BIT(mmMMU_UP_SPI_MASK) &
+		 PROT_BIT(mmMMU_UP_PAGE_ERROR_CAPTURE) &
+		 PROT_BIT(mmMMU_UP_PAGE_ERROR_CAPTURE_VA) &
+		 PROT_BIT(mmMMU_UP_ACCESS_ERROR_CAPTURE) &
+		 PROT_BIT(mmMMU_UP_ACCESS_ERROR_CAPTURE_VA) &
+		 PROT_BIT(mmMMU_UP_RAZWI_WRITE_VLD) &
+		 PROT_BIT(mmMMU_UP_RAZWI_WRITE_ID) &
+		 PROT_BIT(mmMMU_UP_RAZWI_READ_VLD) &
+		 PROT_BIT(mmMMU_UP_RAZWI_READ_ID));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0xF80 */
@@ -893,26 +893,14 @@
 {
 	u32 mask, prot_base, base = mmSTLB_BASE & ~HL_CFG_BASE;
 
-	/* blocking regs within offsets: 0 - 0x80 */
+	/* keep only regs used by LKD: offsets: 0 - 0x80 */
 	prot_base = base;
-	mask = (PROT_BIT(mmSTLB_STLB_FEATURE_EN) &
-		PROT_BIT(mmSTLB_STLB_AXI_CACHE) &
-		PROT_BIT(mmSTLB_HOP_CONFIGURATION) &
-		PROT_BIT(mmSTLB_LINK_LIST) &
-		PROT_BIT(mmSTLB_MEM_CACHE_CONFIG) &
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP0) &
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP1) &
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP2) &
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP3) &
-		PROT_BIT(mmSTLB_SET_THRESHOLD_HOP4));
+	mask = ~(PROT_BIT(mmSTLB_CACHE_INV) &
+		 PROT_BIT(mmSTLB_INV_PS) &
+		 PROT_BIT(mmSTLB_INV_SET));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	/* leaving 1 reg unblocked within offsets: 0x80 - 0x100 */
-	prot_base += SNGL_REG_PROT_SIZE;
-	mask = ~(PROT_BIT(mmSTLB_MEM_READ_ARPROT));
-	regmap_writel(mask, PRIV_PROT_REG(prot_base));
-
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x100 - 0xF80 */
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0xF80 */
 	set_prot_bits_range_all(prot_base, PROT_SIZE_MAX - (prot_base - base));
 }
 
